### PR TITLE
Dx 446-Release-1.0.4/Terms-And-Conditions

### PR DESCRIPTION
**Pull request by W3stside**
Adds T & C's

1. adds new routes required to be accessible BEFORE app load inside disclaimer:
  a. `Cookies`
  b. `HowItWorks`
  c. `Privacy`
2. T & Cs route added - Privacy re-routes to home until its available
3. Selector conditional checks if `provider` / `activeProvider` not yet avail - hacky workaround logic dont like it
4. `WalletIntegration` + `AppValidator` use validation prop (disclaimer_accepted) to skip mount logic if met, also dont like it but prevents app from loading all data etc if user checks out internal routes from `Disclaimer` before accepting

### CODE REVIEW

File: src/utils/index.ts [hunk 1]
```diff
@@ -178,3 +178,25 @@ export const provider2SVG = (providerName: ProviderName | ProviderType) => {
...
   }
 }
+
+export const web3CompatibleNetwork = () => {
+  if (typeof window === 'undefined' || !window.web3) return 'UNKNOWN'
+
+  let netID
+
+  // 1.X.X API
+  if (typeof window.web3.version === 'string') {
```
Comment (Velenir) [inline src/utils/index.ts:188]: Should this be `typeof window.web3.version !== 'string`?

File: src/utils/index.ts [hunk 2]
```diff
@@ -178,3 +178,25 @@ export const provider2SVG = (providerName: ProviderName | ProviderType) => {
...
+    window.web3.eth.net.getId((err: Error, res: string) => {
+      if (err) {
```
Comment (Velenir) [inline src/utils/index.ts]: This has to be a Promise, cause callback.

File: src/components/Footer/index.tsx [hunk 3]
```diff
@@ -25,7 +25,7 @@ const Footer = ({ network }: FooterProps) =>
                 </>
                     :
                 <>
-                    <Link to="/privacy">Privacy Policy</Link><Link to="/cookies">Cookies</Link><Link to="/terms">Terms of Use</Link><Link to="/imprint">Imprint</Link> <span> Protocol built by <img src={gnosisLogoSVG} className="footerLogo"/></span>
+                    <Link to="/privacy">Privacy Policy</Link><Link to="/cookies">Cookies</Link><Link to="terms">Terms and Conditions</Link><Link to="/imprint">Imprint</Link> <span> Protocol built by <img src={gnosisLogoSVG} className="footerLogo"/></span>
```
Comment (Velenir) [inline src/components/Footer/index.tsx]: `/terms` ?

**Review by Velenir — CHANGES_REQUESTED** [review]

File: src/utils/index.ts (hunk 1 above)
Comment (W3stside) [reply src/utils/index.ts:188]: nay

File: src/components/Footer/index.tsx (hunk 3 above)
Comment (W3stside) [reply src/components/Footer/index.tsx]: yes, good catch!

File: src/utils/index.ts (hunk 2 above)
Comment (W3stside) [reply src/utils/index.ts]: _cries_

**Review by Velenir — APPROVED** [review]

**Pull request merged**